tests/TestCase/Console/ConsoleOptionParserTest.php

```diff
@@ -694,6 +693,75 @@ public function testParseArgumentZero(): void
         $this->assertEquals($expected, $result[1], 'Arguments are not as expected');
     }
 
+    /**
+     * Test parsing argument with separator
+     */
+    public function testParseArgumentWithSeparator(): void
+    {
+        $parser = new ConsoleOptionParser('test', false);
+        $parser->addArgument('colors', [
+            'separator' => ',',
+            'choices' => ['red', 'blue', 'green'],
+        ]);
+        $result = $parser->parse(['blue,red'], $this->io);
+        $this->assertEquals([['blue', 'red']], $result[1]);
+    }
+
+    /**
+     * test parse with multiples and default list separator.
+     */
+    public function testParseOptionWithMultiplesDefaultSeparator(): void
+    {
+        $parser = new ConsoleOptionParser('test', false);
+        $parser->addOption('colors', [
+            'multiple' => true,
+            'choices' => ['blue,red', 'yellow,green'],
+        ]);
+        $out = new StubConsoleOutput();
+
+        $result = $parser->parse(['--colors', 'blue,red']);
+        $this->assertEquals(['colors' => ['blue,red'], 'help' => false], $result[0]);
+        $this->assertCount(0, $out->messages());
+    }
+
+    /**
+     * test parse with multiples and custom list separator.
+     */
+    public function testParseOptionWithMultiplesCustomSeparator(): void
+    {
+        $parser = new ConsoleOptionParser('test', false);
+        $parser->addOption('colors', [
+            'multiple' => true,
+            'choices' => ['1st,choice', '2nd,choice', '3rd,choice'],
+            'separator' => ';',
+        ]);
+        $out = new StubConsoleOutput();
+        $io = new ConsoleIo($out, new StubConsoleOutput(), new StubConsoleInput([]));
+
+        $result = $parser->parse(['--colors', '2nd,choice;3rd,choice'], $io);
```

Comment:
Why do we need comma/separator delimited option values when we already have the ability to provide an option multiple times. Like this could be expressed today as `--colors 2nd,choice --colors 3rd,choice`

Reply:
This test check the ability to add `,` in values and separate values by a `;` (custom separator).
There is no default separator anymore, it creates a regression.

Multiple option is kepted (add --option multiple times) and `separator` add a new manner to pass option multiples times. Both are compatible.

Reply:
> Multiple option is kepted (add --option multiple times) and separator add a new manner to pass option multiples times. Both are compatible.

My question was more *why* add separator delimited values when we already have support for list values via multiple options? Having two ways to define list options seems redundant.

Reply:
https://github.com/cakephp/cakephp/issues/17588

Reply:
The current solution is magic switching from string to array and not configured as expectation 
Also unnecessarily verbose and not user friendly.

The new approach allows for Also consistent way for both option and argument

Reply:
> Also unnecessarily verbose and not user friendly.

I think that more verbose forms are useful in scripting or systems administration. You could make multiple options have a short form to reduce the verbosity.

> The current solution is magic switching from string to array and not configured as expectation

That's fair, that was a mistake. In hindsight we should have only returned the first element and required a different method to get all values :shrug:

I'm willing to accept that multiple parameters vs delimited value is a taste thing and that there is some utility in the symmetry for arguments and option lists. :+1: